### DISCUSSION TITLE
&nbsp; appearing before ? and !

### DISCUSSION THREAD

**Discussion by khalwat** [mundschenk-at/php-typography#157] [General]
Using the default setup, it appears that a `&nbsp;` is being inserted before ? and ! characters. What is the setting to disable this?

**Comment by mundschenk-at** [comment]
French punctuation spacing, i.e. `set_french_punctuation_spacing( false )`. The default settings are really not very useful (the API of the methods is consistent, but this produces a weird combination of default values overall).

**Comment by khalwat** [comment]
Yep, I just found it, thanks! 

```
	/**
	 * Enables/disables extra whitespace before certain punction marks, as is the French custom.
	 *
	 * @param bool $on Optional. Default true.
	 */
	public function set_french_punctuation_spacing( $on = true ) {
		$this->data['frenchPunctuationSpacing'] = $on;
	}
```

I'd assume this would only be on by default if the language was French?

**Comment by mundschenk-at** [comment]
No, all the boolean setters default to on. Maybe I should change that in the next major release (semver restrictions). It really makes no sense.
